pkg/anchor/writer/writer.go

```diff
@@ -558,39 +565,108 @@ func (c *Writer) getWitnesses(refs []*operation.Reference) ([]string, error) {
 	return witnesses, nil
 }
 
-func (c *Writer) resolveAnchorOriginViaIPNS(anchorOrigin string) (string, error) {
-	anchorOriginSplitBySlashes := strings.Split(anchorOrigin, "//")
+func (c *Writer) resolveAnchorOriginViaIPNS(anchorOriginIPNSAddressWithScheme, suffix string) (string, error) {
+	var finalAnchorOrigin string
+
+	var ipnsAddressChain []string
+
+	currentAnchorOriginIPNSAddressWithScheme := anchorOriginIPNSAddressWithScheme
+
+	for i := 0; i < maxIPNSResolveLevels; i++ {
+		anchorOriginIPNSAddress := getJustIPNSAddressWithoutScheme(currentAnchorOriginIPNSAddressWithScheme)
+
+		ipnsAddressChain = append(ipnsAddressChain, anchorOriginIPNSAddress)
+
+		webFingerResp, err := c.resolveWebFingerResponseFromIPNS(anchorOriginIPNSAddress)
+		if err != nil {
+			return "", fmt.Errorf("failed to resolve WebFinger response from IPNS. "+
+				"IPNS address chain: %v. Error: %w", ipnsAddressChain, err)
+		}
+
+		origin, err := getProperty(webFingerResp, discoveryrest.OriginType)
+		if err != nil {
+			return "", fmt.Errorf("failed to get origin from WebFinger response. IPNS address chain: %v. "+
+				"Error: %w", ipnsAddressChain, err)
+		}
+
+		originEndpointForSuffix := fmt.Sprintf("%s/%s", origin, suffix)
 
-	logger.Debugf("Resolving anchor origin from %s", anchorOrigin)
+		newAnchorOriginIPNSAddress, err := c.resolveAnchorOriginIPNSAddressFromOriginEndpoint(originEndpointForSuffix)
+		if err != nil {
+			return "", fmt.Errorf("failed to resolve anchor origin from origin endpoint. "+
+				"IPNS address chain: %v. Error: %w", ipnsAddressChain, err)
+		}
+
+		if newAnchorOriginIPNSAddress == anchorOriginIPNSAddressWithScheme {
+			finalAnchorOrigin, err = getProperty(webFingerResp, discoveryrest.WitnessType)
+			if err != nil {
+				return "", fmt.Errorf("failed to get witness property from WebFinger response. "+
+					"IPNS address chain: %v. Error: %w", ipnsAddressChain, err)
+			}
+
+			logger.Debugf("Successfully resolved anchor origin %s from %s. IPNS address chain: %v.",
+				finalAnchorOrigin, anchorOriginIPNSAddressWithScheme, ipnsAddressChain)
+
+			break
+		} else {
+			logger.Debugf("Resolved anchor origin IPNS address %s from %s, but this doesn't match the original "+
+				"anchor origin IPNS address (%s). %s will be resolved next if the max number of resolve levels "+
+				"hasn't been hit yet. Current IPNS address chain: %v.",
+				newAnchorOriginIPNSAddress, currentAnchorOriginIPNSAddressWithScheme,
+				anchorOriginIPNSAddressWithScheme, newAnchorOriginIPNSAddress, ipnsAddressChain)
+			currentAnchorOriginIPNSAddressWithScheme = newAnchorOriginIPNSAddress
```

Comment:
you need to handle if origin return not ipns. it's possible HTTPS

Reply:
@DRK3 https://github.com/hyperledger/aries-framework-go-ext/pull/132#discussion_r652164974